include/openthread/ip6.h

```diff
@@ -391,6 +391,20 @@ otMessage *otIp6NewMessage(otInstance *aInstance, const otMessageSettings *aSett
  */
 typedef void (*otIp6ReceiveCallback)(otMessage *aMessage, void *aContext);
 
+/**
+ * This function pointer is called when an internal IPv6 address is added or removed.
+ *
+ * @param[in]   aAddress            A pointer to the IPv6 address.
+ * @param[in]   aPrefixLength       The prefix length if @p aAddress is unicast address, and 128 for multicast address.
+ * @param[in]   aIsAdded            TRUE if the @p aAddress was added, FALSE if @p aAddress was removed.
+ * @param[in]   aContext            A pointer to application-specific context.
+ *
+ */
+typedef void (*otIp6AddressCallback)(const otIp6Address *aAddress,
```

Comment:
Suggest: moving this next to `otIp6SetAddressCallback` definition?

(looks like other callback `typedef`s in this file are next to their related function declarations and not at the top of the file).